docs/source/tutorials/introductory/logging-first-model/step5-synthetic-data.rst

```diff
@@ -0,0 +1,124 @@
+Create a dataset about apples
+=============================
+
+In order to produce some meaningful data (and a model) for us to log to MLflow, we'll need a dataset.
+In the interests of sticking with our theme of modeling demand for produce sales, this data will
+actually need to be about apples.
+
+There's a distinctly miniscule probability of finding an actual dataset on the internet about this,
+so we can just roll up our sleeves and make our own.
+
+Defining a dataset generator
+----------------------------
+
+For our examples to work, we're going to need something that can actually fit, but not something that
+fits too well. We're going to be training multiple iterations in order to show the effect of modifying
+our model's hyperparameters, so there needs to be some amount of unexplained variance in the feature set.
+However, we need some degree of correlation between our target variable (``demand``, in the case of our
+apples sales data that we want to predict) and the feature set.
+
+We can introduce this correlation by crafting a relationship between our features and our target.
+The random elements of some of the factors will handle the unexplained variance portion.
+
+.. code-section::
+    .. code-block:: python
+
+        import pandas as pd
+        import numpy as np
+        from datetime import datetime, timedelta
+
+
+        def generate_apple_sales_data_with_promo_adjustment(
```

Comment:
Can we use the iris dataset?

Reply:
The iris data set doesn't work well due to the propensity for models to 'memorize' the features, reducing the utility of generating realistic plots, loss values, and makes hyperparameter tuning use cases pointless. We need a dataset that is free, has a distinct amount of covariance, has varying degrees of causal interactivity between the features and the target and among the features themselves in order to meet the needs of both the tutorial and the guide for hyperparameter tuning. 
Datasets of sufficient size that exhibit these traits are either not free to use (UCI datasets with restrictive licenses) or have significant issues with their content (missing / malformed data) or require fetching from a location like kaggle (which is not openly free to download from without an account). 
Because of all of these reasons, a hand-crafted synthetic dataset is the best approach to meet all of the demonstration and tutorial needs.

Reply:
I think it's fine to generate unrealistic or pointless plots/models as long as we can demonstrate what MLlfow is capable of.

Reply:
It doesn't have to be iris. Any simple dataset is ok.